### PR TITLE
complete authentication flow and email verification link implementation

### PR DESCRIPTION
**Pull request by akinwale**

### CODE REVIEW

File: app/src/component/AppNavigator.js [hunk 1]
```diff
@@ -122,6 +131,13 @@ export const AppNavigator = new StackNavigator({
 class AppWithNavigationState extends React.Component {
   static supportedDisplayTypes = ['toast'];
 
+  constructor() {
+    super();
+    this.state = {
```
Comment (skhameneh) [inline app/src/component/AppNavigator.js]: You can define this without a constructor
https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599

File: app/src/constants.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 const Constants = {
-    SETTING_ALPHA_UNDERSTANDS_RISKS: "ALPHA_UNDERSTANDS_RISKS"
+    KEY_FIRST_RUN_EMAIL: "firstRunEmail",
```
Comment (skhameneh) [inline app/src/constants.js]: Why is this value camel case unlike the others?

File: app/src/page/firstRun/internal/email-collect-page.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { Lbry } from 'lbry-redux';
+import {
+  ActivityIndicator,
+  AsyncStorage,
+  Linking,
+  Text,
+  TextInput,
+  View
+} from 'react-native';
+import Button from '../../../component/button';
```
Comment (skhameneh) [inline app/src/page/firstRun/internal/email-collect-page.js]: No aliases (or resolves) for paths?
*Remove the long relative paths

File: app/src/page/firstRun/internal/email-collect-page.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import Colors from '../../../styles/colors';
+import firstRunStyle from '../../../styles/firstRun';
+
+class EmailCollectPage extends React.PureComponent {
+  constructor() {
+    super();
+    this.state = {
```
Comment (skhameneh) [inline app/src/page/firstRun/internal/email-collect-page.js]: Don’t need a full constructor (see comment above)

File: app/src/page/firstRun/internal/email-collect-page.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      email: null,
+      authenticationStarted: false,
+      authenticationFailed: false
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (skhameneh) [inline app/src/page/firstRun/internal/email-collect-page.js]: This is a legacy lifecycle method, should use `getDerivedStateFromProps`

File: app/src/page/firstRun/view.js [hunk 6]
```diff
@@ -47,6 +56,21 @@ class FirstRunScreen extends React.PureComponent {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
```
Comment (skhameneh) [inline app/src/page/firstRun/view.js]: Legacy method, see comment above

File: app/src/page/firstRun/view.js [hunk 7]
```diff
@@ -60,36 +84,92 @@ class FirstRunScreen extends React.PureComponent {
...
-      // Navigate to the splash screen
-      this.launchSplashScreen();
-    } else {
-      // TODO: Page transition animation?
-       this.state.currentPage = FirstRunScreen.pages[pageIndex + 1];
+  handleEmailCollectPageContinue() {
+    const { notify, addUserEmail } = this.props;
+    // validate the email
+    AsyncStorage.getItem(Constants.KEY_FIRST_RUN_EMAIL).then(email => {
+      if (!email || email.trim().length === 0 || email.indexOf('@') === -1) {
```
Comment (skhameneh) [inline app/src/page/firstRun/view.js]: Nice and simple 👍 
There’s an RFC conforming regex I know of, I’ll try to pull it up tomorrow

File: app/src/page/splash/view.js [hunk 8]
```diff
@@ -52,6 +55,49 @@ class SplashScreen extends React.PureComponent {
     });
   }
 
+  componentWillUpdate(nextProps) {
```
Comment (skhameneh) [inline app/src/page/splash/view.js]: Legacy lifecycle method, see other comments about this

File: app/src/constants.js (hunk 2 above)
Comment (akinwale) [reply app/src/constants.js]: The value for the `SETTING_*` key should actually be camel case.

**Pull request merged**